dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ReturnTypesResolver.java

```diff
@@ -44,7 +44,7 @@ public ReturnTypesResolver(ResolverContext context) {
                 .deserializerFactory(maybeAcceptDeserializerFactory
                         .map(TypeName::get)
                         .orElseGet(() -> context.getTypeName(Json.class)))
-                .deserializerFieldName(endpointName.get() + "Deserializer")
+                .deserializerFieldName(InstanceVariables.joinCamelCase(endpointName.get(), "Deserializer"))
```

Comment:
Out of curiosity, is `endpointName + "Deserializer"` ever incorrect?

Reply:
No. But for completeness, might as well have nice names. I don't think I validate endpoint names right now, I guess we have palantir-java-format so it should be guaranteed? Doesn't hurt to do this.